### PR TITLE
PROD-694 Performance improved - activity loading time reduced when multiple attached document

### LINKED ISSUE #3137: Performance issue - Activity item with attached documents taking way longer to load when symlink is disabled
**Describe the bug**
Activity item with multiple attached documents taking way longer to load. This issue is happening when symlink is disabled. See the video: https://www.loom.com/share/f899013a1ce9483da2efca147c9199c9

**To Reproduce**
1. Create an activity item and add multiple( 10 ) image/pdf as document.
2. Make sure you have symlink is disabled
3. Go to the news feed page

**Environment**
demo, dev, local

**Support ticket links**
https://secure.helpscout.net/conversation/1663005486/0?folderId=5218799

**Jira issue** : [PROD-694]

[PROD-694]: https://buddyboss.atlassian.net/browse/PROD-694?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3137 .

### How to test the changes in this Pull Request:

1. Create an activity item and add multiple( 10 ) image/pdf as document.
2. Make sure you have symlink is disabled
3. Go to the news feed page
4. See it is loading fast.

### Proof Screenshots or Video

https://www.loom.com/share/ea9c7664206e4d71845703c77ba539dc

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]

File: src/bp-document/bp-document-functions.php [hunk 1]
```diff
@@ -4238,12 +4238,12 @@ function bp_document_get_preview_url( $document_id, $attachment_id, $size = 'bb-
...
 			}
 
-			if ( $file && ! empty( $file['file'] ) && ( ! empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' !== $extension ) {
+			if ( $file && ! empty( $file['file'] ) && ( empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' !== $extension ) {
 				// Regenerate attachment thumbnails.
 				bb_document_regenerate_attachment_thumbnails( $attachment_id );
 				$file      = image_get_intermediate_size( $attachment_id, $size );
 				$file_path = $file_path . '/' . $file['file'];
-			} elseif ( $file && ! empty( $file['file'] ) && ( ! empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' === $extension ) {
+			} elseif ( $file && ! empty( $file['file'] ) && ( empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' === $extension ) {
```
Comment (dipskakadiya) [inline src/bp-document/bp-document-functions.php:4246]: @mahedihasannoman why both if conditions is same. Can you check once again.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-document/bp-document-functions.php (hunk 1 above)
Comment (mahedihasannoman) [reply src/bp-document/bp-document-functions.php:4246]: Hi @dipskakadiya , Both are not same. Difference is 'pdf' !== $extension & 'pdf' === $extension

Comment (dipskakadiya) [reply src/bp-document/bp-document-functions.php:4246]: @mahedihasannoman I think in that case common condition add as a parent if and we should add inner if. For good code quality. just a structure change.

Comment (mahedihasannoman) [reply src/bp-document/bp-document-functions.php]: Hi @dipskakadiya , Condition modified according to your suggestion.

File: src/bp-document/bp-document-functions.php [hunk 2]
```diff
@@ -4238,15 +4238,17 @@ function bp_document_get_preview_url( $document_id, $attachment_id, $size = 'bb-
...
-				bp_document_generate_document_previews( $attachment_id );
-				$file      = image_get_intermediate_size( $attachment_id, $size );
-				$file_path = $file_path . '/' . $file['file'];
+			if ( $file && ! empty( $file['file'] ) && ( empty( $file['path'] ) || ! file_exists( $file_path ) ) ) {
+				if ( 'pdf' !== $extension ) {
+					// Regenerate attachment thumbnails.
+					bb_document_regenerate_attachment_thumbnails( $attachment_id );
+					$file      = image_get_intermediate_size( $attachment_id, $size );
+					$file_path = $file_path . '/' . $file['file'];
+				} elseif ( 'pdf' === $extension ) {
```
Comment (dipskakadiya) [inline src/bp-document/bp-document-functions.php:4247]: @mahedihasannoman I think else past will be sufficient here not needed else if.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-document/bp-document-functions.php (hunk 2 above)
Comment (mahedihasannoman) [reply src/bp-document/bp-document-functions.php]: Done

**Review by dipskakadiya — DISMISSED** [review]

**Review by chetansatasiya — APPROVED** [review]

**Pull request merged**